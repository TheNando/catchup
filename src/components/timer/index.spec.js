import { h } from 'preact'
import { Provider } from 'preact-redux'
import configureMockStore from 'redux-mock-store'
import { deep, shallow } from 'preact-render-spy'
import ConnectedTimer, { Timer } from './index'

const projects = { projects: ['1', '2'] }
const store = configureMockStore()({ ...projects, dialog: {} })

describe('Timer', () => {
  it('renders the component with a default start', () => {
    const context = deep(<Timer {...projects} />)
    expect(context.output()).toMatchSnapshot()
  })

  it('shows the start time', () => {
    let context = deep(<Timer start="00:05" {...projects} />)
    expect(context.find('#remaining').text()).toBe('0:05')

    context = deep(<Timer start="50:45" {...projects} />)
    expect(context.find('#remaining').text()).toBe('50:45')
  })

  it('sets the selected project', () => {
    const context = deep(<Timer start="00:05" {...projects} />)
    const mockEvent = { target: { selectedIndex: 2 } }
    expect(context.state().projectIndex).toEqual(1)
    context.find('select').simulate('change', mockEvent)
    expect(context.state().projectIndex).toEqual(2)
  })
})

describe('ConnectedTimer', () => {
  it('receives projects from provider', () => {
    const context = deep(
      <Provider store={store}>
        <ConnectedTimer />
      </Provider>
    )
    expect(context.output()).toMatchSnapshot()
  })
})
