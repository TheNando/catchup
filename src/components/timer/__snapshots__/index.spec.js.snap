// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedTimer receives projects from provider 1`] = `
<div class="content">
  <div
    id="remaining"
    class="timer"
  >
    25:00
  </div>
  <select
    class="select"
    hintText="Project"
    outlined={true}
    box={false}
    selectedIndex={1}
    onChange={[Function anonymous]}
  >
    <option>1</option>
    <option>2</option>
  </select>
  <div class="actions">
    <button>
      <i class="icon-button-large">highlight_off</i>
    </button>
    <button>
      <i class="icon-button-large">play_circle_outline</i>
      <i
        on={true}
        class="icon-button-large"
      >
        pause_circle_outline
      </i>
    </button>
    <button>
      <i class="icon-button-large">check_circle_outline</i>
    </button>
  </div>
</div>
`;

exports[`Timer renders the component with a default start 1`] = `
<div class="content">
  <div
    id="remaining"
    class="timer"
  >
    25:00
  </div>
  <select
    class="select"
    hintText="Project"
    outlined={true}
    box={false}
    selectedIndex={1}
    onChange={[Function anonymous]}
  >
    <option>1</option>
    <option>2</option>
  </select>
  <div class="actions">
    <button>
      <i class="icon-button-large">highlight_off</i>
    </button>
    <button>
      <i class="icon-button-large">play_circle_outline</i>
      <i
        on={true}
        class="icon-button-large"
      >
        pause_circle_outline
      </i>
    </button>
    <button>
      <i class="icon-button-large">check_circle_outline</i>
    </button>
  </div>
</div>
`;
